### PR TITLE
Add missing thread safety check in `LoadComponentsAsync`

### PR DESCRIPTION
**Pull request by peppy**
Have run against osu! tests suite, looks okay.

### CODE REVIEW

**Review by smoogipoo — APPROVED** [review]

File: osu.Framework/Graphics/Containers/CompositeDrawable.cs [hunk 1]
```diff
@@ -143,6 +143,8 @@ protected internal Task LoadComponentAsync<TLoadable>([NotNull] TLoadable compon
             if (game == null)
                 throw new InvalidOperationException($"May not invoke {nameof(LoadComponentAsync)} prior to this {nameof(CompositeDrawable)} being loaded.");
 
+            EnsureMutationAllowed(nameof(LoadComponentsAsync));
```
Comment (bdach) [inline osu.Framework/Graphics/Containers/CompositeDrawable.cs:146]: The exception message that results from this is a little funny:

> Cannot mutate the LoadComponentsAsync of a Loaded Drawable while not on the update thread. Consider using Schedule to schedule the mutation operation.

Probably could use a bit more polish.

Comment (peppy) [reply osu.Framework/Graphics/Containers/CompositeDrawable.cs]: Should be better now, well spotted.

File: osu.Framework/Graphics/Containers/CompositeDrawable.cs [hunk 2]
```diff
@@ -143,6 +143,9 @@ protected internal Task LoadComponentAsync<TLoadable>([NotNull] TLoadable compon
             if (game == null)
                 throw new InvalidOperationException($"May not invoke {nameof(LoadComponentAsync)} prior to this {nameof(CompositeDrawable)} being loaded.");
 
+            ThreadSafety.IsUpdateThread = false;
```
Comment (smoogipoo) [inline osu.Framework/Graphics/Containers/CompositeDrawable.cs:146]: Unintended, yeah?

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Framework/Graphics/Containers/CompositeDrawable.cs (hunk 2 above)
Comment (peppy) [reply osu.Framework/Graphics/Containers/CompositeDrawable.cs:146]: 😳

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**